### PR TITLE
[Pillar] Add optional link

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Provide an optional link to the `Pillar` component. 

## List of notable changes:

- **added** Link to the composable API
- **added** tests, stories and documentation
- **added** usage Pillar vs Card guidelines 

## What should reviewers focus on?

- Storybook: https://primer-df9eea3f91-26139705.drafts.github.io/storybook/?path=/story/components-pillar-features--link
- Documentation Usage: https://primer-df9eea3f91-26139705.drafts.github.io/components/Pillar
- Documentation React: https://primer-df9eea3f91-26139705.drafts.github.io/components/Pillar/react

## Steps to test:

1. Open the # component in CI-deployed preview environment
2. Go to # story in Storybook
3. Verify that # behaves as described in the following issue.

## Supporting resources (related issues, external links, etc):

- Issue: https://github.com/github/primer/issues/2280
- Figma: https://www.figma.com/file/9Z6awvfznZj1XmyHYQ2y1z/josepmartins%2Fadd-pillar-link?type=design&node-id=6849-31432&mode=design&t=hFy3WR5Bb6U7xOqG-4

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: apps/docs/content/components/Pillar/index.mdx [hunk 1]
```diff
@@ -56,6 +56,36 @@ When presenting more than four pillars, we recommend displaying them in multiple
 
 ![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)
 
+### Pillar and Card
+
+Pillar and Card components might lead to ambiguity when deciding which one to use, as both could be used to provide pieces of related and condensed information. Some of the differences between them are:
+
+#### Interaction
+
+Cards should be used as navigational items to lead to the extend of the information, the link is always required. Pillars could be used for just plain, non-navigational content or include an optional link and lead to extra information.
```
Comment (josepmartins) [inline apps/docs/content/components/Pillar/index.mdx:65]:
```suggestion
Pillars are used mainly for plain content and might include an option link for extended information. On the other hands, Cards are primarily navigational elements.
```

File: packages/react/src/Pillar/Pillar.tsx [hunk 2]
```diff
@@ -105,12 +116,23 @@ const PillarDescription = forwardRef<HTMLParagraphElement, PillarDescriptionProp
   }
 )
 
+type PillarLinkProps = Omit<LinkProps, 'size' | 'direction'> & BaseProps<HTMLAnchorElement>
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.tsx:119]: `href` is called out as required in the docs, so could you make it required here too please? 

```suggestion
type PillarLinkProps = {
  href: string
} & Omit<LinkProps, 'size' | 'direction'> &
  BaseProps<HTMLAnchorElement>
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 3]
```diff
@@ -54,7 +54,37 @@ If displaying an icon, do so for all cards to achieve a visually balanced layout
 
 When presenting more than four pillars, we recommend displaying them in multiple rows. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns and balancing the number of pillars on each row.
 
-![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)
+![An image showing a group of numerous Pillars stacked in a 3 column grid.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)
+
+### Pillar and Card
+
+Pillar and Card components might lead to ambiguity when deciding which one to use, as both could be used to provide pieces of related and condensed information. Some of the differences between them are:
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:61]: Suggestion to remove the assumption around ambiguity.

```suggestion
Pillar and Card are both components that are used to present related and/or condensed information. They have distinct use-cases however, with the main differences between them being:
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 4]
```diff
@@ -54,7 +54,37 @@ If displaying an icon, do so for all cards to achieve a visually balanced layout
...
+
+#### Interaction
+
+Pillars are used mainly for plain content and might include an optional link for extended information. On the other hand, Cards are primarily navigational elements.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:65]: Could we elaborate on what plain here means? Do you mean non-interactive? 

```suggestion
Pillars are used mainly for non-interactive content and might include an optional link for extended information. On the other hand, Cards are primarily navigational elements.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 5]
```diff
@@ -54,7 +54,37 @@ If displaying an icon, do so for all cards to achieve a visually balanced layout
...
+
+#### Hierarchy
+
+Cards should be used for noteworthy information to draw users attention. Pillars, in the other hand, should be used for common, less noticeable information compared to Cards.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:69]: Alternative wording suggestion. I wasn't sure what 'common' meant here, but think lower priority could be a good alternative? LMK if i'm off-base here.

```suggestion
Cards should be used for noteworthy information to immediately draw user attention. Pillars, on the other hand, should be used for lower priority information compared to Cards.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 6]
```diff
@@ -54,7 +54,37 @@ If displaying an icon, do so for all cards to achieve a visually balanced layout
...
+
+#### Group size
+
+Cards should be used in small groups (3, 4 or 5 items maximum) whereas Pillars might be used on both small and big groups (more than 5 items). The more items a users see's within a group, the less noticeable they become as unique elements.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:73]: Would you ever want to use cards in a group of 5? 

```suggestion
Cards should be used in small groups (3 or 4 maximum), whereas Pillars can be used in larger or smaller groups (5+). The more items a users sees within a group, the less noticeable they become as unique elements.
```

Comment (rezrah) [reply apps/docs/content/components/Pillar/index.mdx:73]: Or slight alternative to the last line..

```suggestion
Cards should be used in small groups (3 or 4 maximum), whereas Pillars can be used in larger or smaller groups (5+). As users encounter an increasing number of items within a group, the distinctiveness of each individual element diminishes.
```

Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:73]: Also a separate question around this bit:

> Pillars might be used on both **small** and big groups

Would you ever want to use Pillars in less than three config? Not sure if we should encourage that as there are better ways to present sparse infomation. Should this guidance be focussed on recommending Pillars for larger quantities of grouped information?

Comment (josepmartins) [reply apps/docs/content/components/Pillar/index.mdx:73]: Love the last suggestion 👍

File: packages/react/src/Pillar/Pillar.module.css [hunk 7]
```diff
@@ -24,10 +24,26 @@
...
   color: var(--brand-color-text-muted);
 }
 
+.Pillar__link {
+  margin-top: var(--base-size-8);
+}
+
+.Pillar__link,
+.Pillar__link span {
+  color: var(--brand-Pillar-link-color);
```
[the remark this answers is not in the document]
Comment (josepmartins) [reply packages/react/src/Pillar/Pillar.module.css]: Made a variant prop with `default`/`accent` options

File: packages/react/src/Pillar/Pillar.test.tsx [hunk 8]
```diff
@@ -82,4 +82,20 @@ describe('Pillar', () => {
     const pillarEl = getByTestId(mockTestId).firstChild
     expect(pillarEl).toHaveClass(classToCheck)
   })
+
+  it('renders the link correctly into the document', () => {
+    const mockTestId = 'test'
+    const classToCheck = 'Pillar__link'
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.test.tsx:88]: Would be cool to also check that the `Link--accent` gets set by default. WDYT?

**Review by rezrah — APPROVED** [review]
Looks great 👏.. just some additional minor, but non-blocking feedback.

**Pull request merged**